src/interface/misc.rs

```diff
@@ -52,6 +58,13 @@ pub fn cagetable_getref(cageid: u64) -> RustRfc<Cage> {
     unsafe { CAGE_TABLE[cageid as usize].as_ref().unwrap().clone() }
 }
 
+pub fn cagetable_getref_opt(cageid: u64) -> Option<RustRfc<Cage>> {
+    unsafe { match CAGE_TABLE[cageid as usize].as_ref() {
+        Some(cage) => Some(cage.clone()),
+        None => None
+    }}
+}
+
 pub fn cagetable_clear() {
```

Comment:
Above code can be simplified as follow:
`unsafe { CAGE_TABLE[cageid as usize].as_ref().map(|cage| cage.clone()) }
`
Here we are using the unsafe block, which means we are bypassing some of Rust's safety checks. In this case, we are indexing into CAGE_TABLE with cageid as usize. We must ensure that cageid is always a valid index (i.e., it's never out of bounds). And make sure len of CAGE_TABLE is not 0.